pkg/updater/version.go

```diff
@@ -131,31 +130,13 @@ const (
 	maxErrorResponseBytes = 2048
 )
 
-// CheckAppVersion calls CheckAppVersionSync in a go routine. It returns a closure
-// to be run after program logic which will block until a response is returned
-// or provided context is canceled. If no provided deadline in context, defaults to 2 seconds.
-// If there is an update, out() will be called during the
-// returned closure.
-//
-// If no update is available: out() will not be called.
-// If there is an error: out() will not be called, message will be logged as WARN.
-// If the context is canceled: out() is not called.
-// Example out(): `func(s string) {fmt.Fprintln(os.Stderr, s)}`.
-func CheckAppVersion(ctx context.Context, params *CheckVersionParams, out func(string)) func() {
-	cancel := func() {}
-	if _, ok := ctx.Deadline(); !ok {
-		ctx, cancel = context.WithTimeout(ctx, time.Second*2)
-	}
-
-	return asyncFunctionCall(ctx, func() (string, error) {
-		defer cancel()
-		return CheckAppVersionSync(ctx, params)
-	}, out)
-}
+// CheckAppVersion checks if a newer version of an app is available. Any
+// relevant update info will be returned as a string. It accepts a context for
+// cancellation, and has a default timeout of 5 seconds.
+func CheckAppVersion(ctx context.Context, params *CheckVersionParams) (string, error) {
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
```

Comment:
This might not be right. If the passed context already has a timeout that is larger than 5 seconds, we shouldn't shorten the timeout.

Reply:
Yeah, I went ahead and merged this pattern though I used to have it respect a longer timeout if it was passed in.

Probably worth ensuring we are all on the same page as to desired behavior.

I don't see 5 second timeout being an issue except for debugging purposes.

Reply:
We could be smarter here, but in practice I think most services will accept the default timeout without customizing it.

I'm also fine not capping the timeout at all (and relying on the caller to set a timeout in the context).